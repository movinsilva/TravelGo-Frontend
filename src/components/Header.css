header {
  position: fixed;
  width: 100%;
  z-index: 1;
}/*# sourceMappingURL=Header.css.map */