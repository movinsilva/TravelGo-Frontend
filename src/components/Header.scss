header {
    position: fixed;
    width: 100%;
    z-index: 1;
}
.logo{
    width: 90px;
    height: auto;
    margin: 0;
    padding: 5px;
}