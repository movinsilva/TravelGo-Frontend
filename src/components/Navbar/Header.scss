header {
    position: fixed;
    width: 100%;
    z-index: 1;
}
.logo{
    width: 90px;
    height: auto;
    margin-right: 5px;
    filter: brightness(1.4)

}
.navContent{
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: auto;
    height: 60px;
    
    .nav-link {
        margin-right: 20px; 
     }
}
