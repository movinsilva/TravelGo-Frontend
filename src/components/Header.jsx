import { Navbar, Nav, Container } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import { FaSignInAlt, FaSignOutAlt } from 'react-icons/fa';
import './Header.scss'
// Import the logo image
import logo from '../assets/LogoBlue.png';

const Header = () => {
  return (
    <header>
      <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
        <Container>
          <Navbar.Brand href="./">
            <img
              alt=""
              src={logo}
              className="logo"
            />{' '}
            Travel Go
          </Navbar.Brand>

          <Navbar.Toggle aria-controls='basic-navbar-nav' />

          <Navbar.Collapse id='basic-navbar-nav'>
            <Nav className='ms-auto'>

              <LinkContainer to='/login'>
                <Nav.Link>
                  <FaSignInAlt /> Sign In
                </Nav.Link>
              </LinkContainer>

              <LinkContainer to='/register'>
                <Nav.Link>
                  <FaSignOutAlt /> Sign Up
                </Nav.Link>
              </LinkContainer>

            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
};

export default Header;