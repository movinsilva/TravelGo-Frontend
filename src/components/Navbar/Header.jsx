import { Navbar, Nav, Container, NavDropdown } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import { FaSignInAlt, FaSignOutAlt } from 'react-icons/fa';
import './Header.scss'
import logo from '../../assets/LogoBlue.png';
import NavLinks from './NavLinks';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';
import { FaUserCircle } from 'react-icons/fa'
import { logout } from '../../slices/authSlice';
import { useLogoutMutation } from '../../slices/usersApiSlice';

const Header = () => {

  const { userInfo } = useSelector((state) => state.auth);
  var isLoggedIn = false;

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const [logoutApiCall] = useLogoutMutation();

  const logoutHandler = async () => {
    try {
      await logoutApiCall().unwrap();
      dispatch(logout());
      navigate('/');
    } catch (err) {
      console.error(err);
    }
  };

  useEffect(() => {
    
    if (userInfo) {
      isLoggedIn = true;
    }
  })

  return (
    <header>
      <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
        <Container fluid={true}>
          <Navbar.Brand href="./">
            <img
              alt=""
              src={logo}
              className="logo"
            />{' '}
            TravelGo
          </Navbar.Brand>
          <Navbar.Toggle aria-controls='basic-navbar-nav' />
          <Navbar.Collapse id='basic-navbar-nav'>
            <Nav className="me-auto navContent">
              <NavLinks />
            </Nav>
            <Nav >

              {userInfo ? (
                // Render user icon when the user is logged in
                <>
                  <NavDropdown title={userInfo.firstname} id='name'>
                    <LinkContainer to='/profile'>
                      <NavDropdown.Item>Profile</NavDropdown.Item>
                    </LinkContainer>
                    <NavDropdown.Item onClick={logoutHandler}>Logout</NavDropdown.Item>
                  </NavDropdown>
                </>
              ) : (
                // Render "Sign In" and "Sign Up" links when the user is not logged in
                <>

                  <LinkContainer to='/login'>
                    <Nav.Link>
                      <FaSignInAlt /> Sign In
                    </Nav.Link>
                  </LinkContainer>

                  <LinkContainer to='/register'>
                    <Nav.Link>
                      <FaSignOutAlt /> Sign Up
                    </Nav.Link>
                  </LinkContainer>

                </>
              )}
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
};

export default Header;