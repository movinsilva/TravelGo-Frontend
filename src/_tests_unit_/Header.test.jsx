import { describe, expect, it, render } from "vitest";
import TestRenderer from "react-test-renderer"; 
import Header from "../components/Navbar/Header";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux"; // Import the Provider
import store from "../store.js"; // Import your Redux store

describe("Header.jsx", () => {
    

    it("Login and logout", () => {
        const tree = TestRenderer.create(
        /* Making the redux store available for all the components */
          <Provider store={store}>   
            <MemoryRouter>
              <Header />
            </MemoryRouter>
          </Provider>
        ).toJSON();

        //login
        const linkcontainer = tree.children[0].children[0].children[2].children[1];
        expect(linkcontainer.children[0].type).toBe("a");
        expect(linkcontainer.children[0].props.href).toBe("/login");

        expect(linkcontainer.children[1].type).toBe("a");
        expect(linkcontainer.children[1].props.href).toBe("/register");
        console.log(linkcontainer)

        
})
    it("renders Header navbar navigation components", () => {
        const tree = TestRenderer.create(
        /* Making the redux store available for all the components */
          <Provider store={store}>   
            <MemoryRouter>
              <Header />
            </MemoryRouter>
          </Provider>
        ).toJSON();
        //navbar navigation links
        const navbarlinks = tree.children[0].children[0].children[2].children[0].children[0];


        console.log(tree.children[0].children[0].children[2].children[0].children[0])
        expect(navbarlinks.children[0].type).toBe("a");
        expect(navbarlinks.children[0].props.href).toBe("/");

        expect(navbarlinks.children[1].type).toBe("a");
        expect(navbarlinks.children[1].props.href).toBe("/Services");

        expect(navbarlinks.children[2].type).toBe("a");
        expect(navbarlinks.children[2].props.href).toBe("/Bookings");
        
        expect(navbarlinks.children[3].type).toBe("a");
        expect(navbarlinks.children[3].props.href).toBe("/Terms");
        
        expect(navbarlinks.children[4].type).toBe("a");
        expect(navbarlinks.children[4].props.href).toBe("/ContactUs");
       

        console.log(navbarlinks)
        expect(tree).toMatchSnapshot(); // This can be used for snapshot testing

    })
});

describe("Header.jsx", () => {
  it("renders Header component", () => {
    const tree = TestRenderer.create(
    /* Making the redux store available for all the components */
      <Provider store={store}>   
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    ).toJSON();
   /* 
    //header
    console.log(tree)
    //container fluid div
    console.log(tree.children[0].children[0])
  */
    

    //nav
    
    expect(tree.children[0].type).toBe("nav");
    expect(tree.children[0].props.className).toBe("nav-style navbar navbar-expand-lg navbar-light");

    
    //routing to homepage
    
    expect(tree.children[0].children[0].children[0].type).toBe("a");
    expect(tree.children[0].children[0].children[0].props.href).toBe("/");


    //navbar toggle
    
    expect(tree.children[0].children[0].children[1].type).toBe("button");
    expect(tree.children[0].children[0].children[1].props["aria-controls"]).toBe("basic-navbar-nav");
    expect(tree.children[0].children[0].children[1].props.type).toBe("button");
    expect(tree.children[0].children[0].children[1].props["aria-label"]).toBe("Toggle navigation");
    
    
    //navbar collapse
    expect(tree.children[0].children[0].children[1].props.className).toBe("navbar-toggler collapsed");

    expect(tree.type).toBe("header");
    expect(tree.children[0].type).toBe("nav");
    


    
    
  });
});

