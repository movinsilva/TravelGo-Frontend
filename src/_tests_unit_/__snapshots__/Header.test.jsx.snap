// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.jsx > Header navbar navigation components 1`] = `
<header>
  <nav
    className="nav-style navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={{}}
      >
        <img
          alt=""
          className="logo"
          src="/src/assets/LogoBlack.png"
        />
         
        TravelGo
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto nav-content navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="me-auto navContent navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="active nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
              style={{}}
            >
              Home
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Services"
              disabled={false}
              href="/Services"
              onClick={[Function]}
              style={null}
            >
              Services
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Bookings"
              disabled={false}
              href="/Bookings"
              onClick={[Function]}
              style={null}
            >
              Bookings
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Terms"
              disabled={false}
              href="/Terms"
              onClick={[Function]}
              style={null}
            >
              Terms & Conditions
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/ContactUs"
              disabled={false}
              href="/ContactUs"
              onClick={[Function]}
              style={null}
            >
              Contact Us
            </a>
          </div>
        </div>
        <div
          className="end-nav navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign In
            </span>
          </a>
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/register"
            disabled={false}
            href="/register"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign Up
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header.jsx > Header navbar navigation components redirects to correct location 1`] = `
<header>
  <nav
    className="nav-style navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={{}}
      >
        <img
          alt=""
          className="logo"
          src="/src/assets/LogoBlack.png"
        />
         
        TravelGo
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto nav-content navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="me-auto navContent navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="active nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
              style={{}}
            >
              Home
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Services"
              disabled={false}
              href="/Services"
              onClick={[Function]}
              style={null}
            >
              Services
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Bookings"
              disabled={false}
              href="/Bookings"
              onClick={[Function]}
              style={null}
            >
              Bookings
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Terms"
              disabled={false}
              href="/Terms"
              onClick={[Function]}
              style={null}
            >
              Terms & Conditions
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/ContactUs"
              disabled={false}
              href="/ContactUs"
              onClick={[Function]}
              style={null}
            >
              Contact Us
            </a>
          </div>
        </div>
        <div
          className="end-nav navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign In
            </span>
          </a>
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/register"
            disabled={false}
            href="/register"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign Up
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header.jsx > renders Header  1`] = `
<header>
  <nav
    className="nav-style navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={{}}
      >
        <img
          alt=""
          className="logo"
          src="/src/assets/LogoBlack.png"
        />
         
        TravelGo
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto nav-content navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="me-auto navContent navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="active nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
              style={{}}
            >
              Home
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Services"
              disabled={false}
              href="/Services"
              onClick={[Function]}
              style={null}
            >
              Services
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Bookings"
              disabled={false}
              href="/Bookings"
              onClick={[Function]}
              style={null}
            >
              Bookings
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Terms"
              disabled={false}
              href="/Terms"
              onClick={[Function]}
              style={null}
            >
              Terms & Conditions
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/ContactUs"
              disabled={false}
              href="/ContactUs"
              onClick={[Function]}
              style={null}
            >
              Contact Us
            </a>
          </div>
        </div>
        <div
          className="end-nav navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign In
            </span>
          </a>
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/register"
            disabled={false}
            href="/register"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign Up
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header.jsx > renders Header component 1`] = `
<header>
  <nav
    className="nav-style navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={{}}
      >
        <img
          alt=""
          className="logo"
          src="/src/assets/LogoBlack.png"
        />
         
        TravelGo
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto nav-content navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="me-auto navContent navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="active nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
              style={{}}
            >
              Home
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Services"
              disabled={false}
              href="/Services"
              onClick={[Function]}
              style={null}
            >
              Services
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Bookings"
              disabled={false}
              href="/Bookings"
              onClick={[Function]}
              style={null}
            >
              Bookings
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Terms"
              disabled={false}
              href="/Terms"
              onClick={[Function]}
              style={null}
            >
              Terms & Conditions
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/ContactUs"
              disabled={false}
              href="/ContactUs"
              onClick={[Function]}
              style={null}
            >
              Contact Us
            </a>
          </div>
        </div>
        <div
          className="end-nav navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign In
            </span>
          </a>
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/register"
            disabled={false}
            href="/register"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign Up
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header.jsx > renders Header navbar navigation components 1`] = `
<header>
  <nav
    className="nav-style navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="active navbar-brand"
        href="/"
        onClick={[Function]}
        style={{}}
      >
        <img
          alt=""
          className="logo"
          src="/src/assets/LogoBlack.png"
        />
         
        TravelGo
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto nav-content navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="me-auto navContent navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="active nav-link"
              data-rr-ui-event-key="/"
              disabled={false}
              href="/"
              onClick={[Function]}
              style={{}}
            >
              Home
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Services"
              disabled={false}
              href="/Services"
              onClick={[Function]}
              style={null}
            >
              Services
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Bookings"
              disabled={false}
              href="/Bookings"
              onClick={[Function]}
              style={null}
            >
              Bookings
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/Terms"
              disabled={false}
              href="/Terms"
              onClick={[Function]}
              style={null}
            >
              Terms & Conditions
            </a>
            <a
              className="nav-link"
              data-rr-ui-event-key="/ContactUs"
              disabled={false}
              href="/ContactUs"
              onClick={[Function]}
              style={null}
            >
              Contact Us
            </a>
          </div>
        </div>
        <div
          className="end-nav navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/login"
            disabled={false}
            href="/login"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign In
            </span>
          </a>
          <a
            className="small-icon-link nav-link"
            data-rr-ui-event-key="/register"
            disabled={false}
            href="/register"
            onClick={[Function]}
            style={null}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              />
            </svg>
             
            <span
              className="nav-link-text"
            >
              Sign Up
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
